send.go

```diff
@@ -101,53 +101,65 @@ func (d *DBFT[H]) sendPrepareResponse() {
 	d.broadcast(msg)
 }
 
-func (c *Context[H]) makePreCommit() ConsensusPayload[H] {
-	if msg := c.PreCommitPayloads[c.MyIndex]; msg != nil {
+func (d *DBFT[H]) makePreCommit() ConsensusPayload[H] {
+	if msg := d.PreCommitPayloads[d.MyIndex]; msg != nil {
 		return msg
 	}
 
-	if preB := c.CreatePreBlock(); preB != nil {
+	if preB := d.CreatePreBlock(); preB != nil {
 		var preData []byte
-		if err := preB.SetData(c.Priv); err == nil {
+		if err := preB.SetData(d.Priv); err == nil {
 			preData = preB.Data()
+		} else {
+			d.Logger.Error("PreCommit data construction failed", zap.Error(err))
+			return nil
 		}
 
-		preCommit := c.Config.NewPreCommit(preData)
+		preCommit := d.Config.NewPreCommit(preData)
 
-		return c.Config.NewConsensusPayload(c, PreCommitType, preCommit)
+		return d.Config.NewConsensusPayload(&d.Context, PreCommitType, preCommit)
 	}
 
 	return nil
 }
 
-func (c *Context[H]) makeCommit() ConsensusPayload[H] {
-	if msg := c.CommitPayloads[c.MyIndex]; msg != nil {
+func (d *DBFT[H]) makeCommit() ConsensusPayload[H] {
```

Comment:
Likely you can keep it bound to `*Context[H]` if you're to return an error which then will be logged in `send*`

Reply:
Good, fixed!